GitUI/BranchTreePanel/RepoObjectsTree.Nodes.Remotes.cs

```diff
@@ -47,20 +58,56 @@ protected override async Task LoadNodesAsync(CancellationToken token)
                         var parent = remoteBranchNode.CreateRootNode(
                             nodes,
                             (tree, parentPath) => CreateRemoteBranchPathNode(tree, parentPath, remote));
+
                         if (parent != null)
                         {
-                            Nodes.AddNode(parent);
+                            enabledRemoteRepoNodes.Add((RemoteRepoNode)parent);
                         }
                     }
                 }
 
+                // Create nodes for enabled remotes without branches
+                var enabledRemotesNoBranches = gitRemoteManager.GetEnabledRemotesNameWithoutBranches();
+                foreach (var remoteName in enabledRemotesNoBranches)
+                {
+                    if (remoteByName.TryGetValue(remoteName, out var remote))
+                    {
+                        var node = new RemoteRepoNode(this, remoteName, gitRemoteManager, remote, true);
+                        enabledRemoteRepoNodes.Add(node);
+                    }
+                }
+
+                // Add enabled remote nodes in order
+                enabledRemoteRepoNodes
+                    .OrderBy(node => node.FullPath)
+                    .ForEach(node => Nodes.AddNode(node));
+
+                // Add disabled remotes, if any
+                var disabledRemotes = gitRemoteManager.GetDisabledRemotes();
+                if (disabledRemotes.Count() > 0)
```

Comment:
`disabledRemotes` is of `IReadOnlyList<T>` type, no need to call an extension method, `.Count` is available.
Suggest inverting `if` to reduce nesting too.

Reply:
Use .Any.

Count iterates over the whole ienumerable and generates a count.  >any returns false if empty and if it finds one item, it stops and returns true.

Reply:
Any also takes a lambda so you can say 

theIenumerable.Any(x=> x.Enabled==false)

Reply:
> Count iterates over the whole ienumerable and generates a count. .Any returns false if empty and if it finds one item, it stops and returns true.

The Count() extension method will only iterate if the underlying type does't already store a Count field; IReadOnlyList does store a Count, so the complexity of Count() is O(1). There's only a slight bit more overhead since the extension method checks what kind of collection it's being called on.

> RussKie: disabledRemotes is of IReadOnlyList<T> type, no need to call an extension method, .Count is available.
> vbjay: Use .Any().

I'm inclined to use Any() since it semantically means Count > 0. Is that okay @RussKie?

> Suggest inverting if to reduce nesting too.

I'm on board for reducing nesting, and totally agree with early outs; but not when it's this deep into the function. Early returns in the middle of a somewhat longer function can be surprising, especially if people start adding code after this. In this specific case, I'd like to leave the condition as is, is that alright?

Reply:
I prefer to call `.Count` if I know the property exists. But I'll settle for `.Any()`, but as it was mentioned it has an extra cost.
Here's a response from [Jared Parson](https://twitter.com/jaredpar) - https://stackoverflow.com/a/4098267/2338036, that guy builds the C# compiler.

Reply:
> I'm on board for reducing nesting, and totally agree with early outs; but not when it's this deep into the function. Early returns in the middle of a somewhat longer function can be surprising

On a contrary, it is absolutely normal to have exit points in a middle of a method - this means a specific code path is finished. It also makes it easier to test.
Arguably the method should not be very long, but that's a different discussion 😏

Reply:
@RussKie Is it a deal breaker for you if I leave it nested as-is? I find it actually hurts readability in this specific case when I invert the condition and add the early out. If you really want it, I'll do it.